src/OpenLoco/IndustryManager.cpp

```diff
@@ -190,18 +190,238 @@ namespace OpenLoco::IndustryManager
+            {
+                if (Map::TileManager::countSurroundingDesertTiles(randomPos) < 100)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtNearDesert)
+            {
+                if (Map::TileManager::countSurroundingDesertTiles(randomPos) >= 70)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtNearWater)
+            {
+                if (Map::TileManager::countSurroundingWaterTiles(randomPos) < 10)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtAwayFromWater)
+            {
+                if (Map::TileManager::countSurroundingWaterTiles(randomPos) != 0)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtOnWater)
+            {
+                auto tile = Map::TileManager::get(randomPos);
+                auto* surface = tile.surface();
+                if (surface != nullptr && surface->water() == 0)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtNearTown)
+            {
+                auto res = TownManager::getClosestTownAndUnk(randomPos);
+                if (!res.has_value())
+                {
+                    continue;
+                }
+                const auto& [townId, builtUp] = *res;
+                if (builtUp == 0)
+                {
+                    const auto* town = TownManager::get(townId);
+                    if (Math::Vector::manhattanDistance(randomPos, Map::Pos2{ town->x, town->y }) > 576)
+                    {
+                        continue;
+                    }
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtAwayFromTown)
+            {
+                auto res = TownManager::getClosestTownAndUnk(randomPos);
+                if (!res.has_value())
+                {
+                    continue;
+                }
+                const auto townId = res->first;
+                const auto* town = TownManager::get(townId);
+                if (Math::Vector::manhattanDistance(randomPos, Map::Pos2{ town->x, town->y }) < 768)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtNearTrees)
+            {
+                if (Map::TileManager::countSurroundingTrees(randomPos) < 25)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtRequiresTrees)
+            {
+                if (Map::TileManager::countSurroundingTrees(randomPos) > 3)
```

Comment:
This seems strange for a flag named `builtRequiresTrees`. More than three trees and we can't build?

Reply:
Perhaps my naming wasn't great. I'll recheck

Reply:
`requiresOpenSpace`? `requiresPlains`?